test/connection.js

```diff
@@ -199,7 +270,11 @@ describe('Hapi server', () => {
         }, (err) => {
 
             expect(err).to.not.exist();
-            expect(server.plugins['hapi-mongodb'].db.options.url).to.equal('mongodb://localhost:27017');
+            const db = server.plugins['hapi-mongodb'].db;
+            expect(db).to.be.instanceof(Mongodb.Db);
+            expect(db.databaseName).to.equal('test');
+            expect(db.topology.s.clonedOptions.host).to.equal('localhost');
```

Comment:
You are right.
The problem is that when using  `mongodb@2.1.20` or above
```javascript
expect(server.plugins['hapi-mongodb'].db.options.url).to.equal('mongodb://localhost:27017');
```
Doesn't pass

So it might be a driver's internal too (it might also be a bug) - I will check that.
We can also just create a different test that fakes mongodb just to check the options are being passed correctly